src/llm/legacy/servable.hpp

```diff
@@ -0,0 +1,56 @@
+//*****************************************************************************
+// Copyright 2025 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+#include <memory>
+#include <string>
+
+#include <openvino/genai/continuous_batching_pipeline.hpp>
+
+#include "../servable.hpp"
+#include "legacy_executor.hpp"
+#include "src/llm/llm_calculator.pb.h"
+
+namespace ovms {    
+
+struct LegacyServableProperties : public GenAiServableProperties {
+    ov::genai::SchedulerConfig schedulerConfig;
+    std::shared_ptr<ov::genai::LLMPipeline> pipeline;
+    std::shared_ptr<LegacyExecutorWrapper> legacyExecutor;
+};
+
+class LegacyServable : public GenAiServable {
+    std::shared_ptr<LegacyServableProperties> properties;
+
+protected:
+    void notifyExecutorThread();
+
+public:
+    LegacyServable() {
+        properties = std::make_shared<LegacyServableProperties>();
+    }
+
+    // Interface methods
+    std::shared_ptr<GenAiServableExecutionContext> createExecutionContext() override;
+    std::shared_ptr<GenAiServableProperties> getProperties() override;
+    absl::Status scheduleExecution(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
+    absl::Status readCompleteExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
+    absl::Status readPartialExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
+    absl::Status prepareCompleteResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
+    absl::Status preparePartialResponse(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
+    bool supportsSpeculativeDecoding() const override;
+    absl::Status parseRequest(std::shared_ptr<GenAiServableExecutionContext>& executionContext) override;
```

Comment:
It's a minor, but can we have a method order matching base class? 
Also the implementation in cpp file should match that order